common/canonical_serialization/src/canonical_serialization_test.rs

```diff
@@ -319,3 +319,47 @@ fn test_deserialization_failure_cases() {
     deserializer = SimpleDeserializer::new(&bool_bytes);
     assert!(deserializer.clone().decode_bool().is_err());
 }
+
```

Comment:
The types we're testing this on (tuples, strings) are amenable to writing a proptest [Strategy](https://docs.rs/proptest/0.9.4/proptest/strategy/index.html) based on their underlying `Arbitrary` impls. Can we do that or at the very least open an issue to do so and mark it as a "good first contribution"?

Reply:
Yes! Frankly, most of this should be rewritten as a proptest. So most definitely a good onboarding contribution.